crates/configuration/src/network.rs

```diff
@@ -52,27 +52,35 @@ states! {
 #[derive(Debug)]
 pub struct NetworkConfigBuilder<State> {
     config: NetworkConfig,
+    errors: Vec<Box<dyn Error>>,
     _state: PhantomData<State>,
 }
 
 impl Default for NetworkConfigBuilder<Initial> {
     fn default() -> Self {
         Self {
             config: NetworkConfig {
-                global_settings: GlobalSettingsBuilder::new().build(),
+                global_settings: GlobalSettingsBuilder::new()
+                    .build()
+                    .expect("no errors for default builder"),
```

Comment:
If we use `expect` here we should ask the users to please open an issue in our repo since if this fails is because of some bug in our code.